docs/scos/dev/keeping-a-project-upgradable/upgradability-guidelines/dependency-provider-contains-addition-logic.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Dependency provider contains addition logic
+description: Reference information for evaluator tools.
+template: howto-guide-template
+---
+
+Additional logic inside the dependency provider’s methods.
+
+## Problem description
+
+On the project level, developers use IF constructs with variety of expressions in dependency providers to register the plugins in particular cases only.
+
+Not all possible expressions are needed inside of the if statements for plugin registration and not all of them are supported. This check will verify that if IF construct is used for plugin registration, then only one of the following expressions is used:
+
+```php
+class_exists(\Foo\Bar::class) function call - it is required for BC reasons
+```
+    
+```php
+$this->getConfig()->isDevelopmentConsoleCommandsEnabled() function calls - it is required for plugins that are needed in development mode only (e.g.: profiling, debug, etc.)
+```
+
+## Example of code that causes an upgradability error:
+
+The method `getFormPlugins` in `FormDependencyProvider` contains unsupported expressions in IF construct `self::IS_DEV`.
+
+```php
+use Spryker\Yves\Form\FormDependencyProvider as SprykerFormDependencyProvider;
+
+class FormDependencyProvider extends SprykerFormDependencyProvider
+{
+    public const IS_DEV = true;
+    
+    ...
+    
+    protected function getFormPlugins(): array
+    {
+        $plugins = [
+            new ValidatorExtensionFormPlugin(),
+        ];
+        
+        $plugins[] = new CsrfFormPlugin();
+        
+        if (self::IS_DEV) {
```

Comment:
`self` not a good example here, because it is not recommended by Spryker's conventions. Please, change it to `static`.

Reply:
++